File: app/src/main/java/com/jedev/leflix/service/API/Requests/GoogleBooksAPI.java
package com.jedev.leflix.service.API.Requests;

import com.jedev.leflix.service.API.Entities.Volume;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface GoogleBooksAPI {

    @GET("volumes?printType=books")
    Call<Volume> searchVolumes(
            @Query("key") String apiKey,
            @Query("langRestrict") String lang,
            @Query("maxResults") String maxResults,
            @Query("q") String termSearch
    );
}

File: app/src/main/java/com/jedev/leflix/service/api/requests/GoogleBooksAPI.kt
package com.jedev.leflix.service.api.requests

import com.jedev.leflix.service.api.entities.Volume

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface GoogleBooksAPI {

    @GET("volumes?printType=books&orderBy=newest")
    fun searchVolumes(
            @Query("key") apiKey: String,
            @Query("langRestrict") lang: String,
            @Query("maxResults") maxResults: Int,
            @Query("q") termSearch: String
    ): Call<Volume>

    @GET("volumes/{volumeId}")
    fun getById(
            @Path("volumeId") volumeId: String,
            @Query("key") apiKey: String
    ): Call<Volume>
}
